### DISCUSSION TITLE
Manually setting extensions.zotero.translators.better-bibtex.citekeyFormat

### DISCUSSION THREAD

**Discussion by okomestudio** [retorquere/zotero-better-bibtex#3005] [General]
I'm using a configuration manager (Ansible) to bootstrap Zotero installation. For each installation, I update the `user.js` file (which influences `prefs.js`) to set the app config settings automatically for my personal configuration.

This works for most settings, but for some reason I cannot get it to update ` extensions.zotero.translators.better-bibtex.citekeyFormat` (and ` extensions.zotero.translators.better-bibtex.citekeyFormatEdit`, for which I'm not sure how they are related). This is the setting for citekey generation formula. It seems to always reset to `"auth.lower + shorttitle(3,3) + year"` no matter how I set this config item.

There is a somewhat similar field `extensions.zotero.translators.better-bibtex.postscript` which takes in JavaScript snippet, which I can manually set without any issue.

Does anyone know how I can set the above field(s) by updating `user.js` (so that `prefs.js` take them in)?

**Comment by retorquere** [comment]
> `extensions.zotero.translators.better-bibtex.citekeyFormatEdit`, for which I'm not sure how they are related).

I'm abusing the preference system to get transaction-like updates for the citekey formula. In Zotero, preference changes are propagated as soon as they change, but with the citekey formula, I only want the preference to change from one valid formula to another, and while you're typing, the formula often isn't valid. So in the prefs pane, you're actually editing `citekeyFormatEdit`, which has one listener, which compiles the formula on every change, and if the compilation succeeds, copies it to `citekeyFormat`. This in turn wakes up the parts of the code that are actually interested in formula changes, which a guarantee they will only be woken for a valid formula.

I'd try just setting both to the same value with your automation tool, after updating to 6.7.239. It has a change that may be relevant.

**Reply by okomestudio** [reply]
Thanks for the reply. Strange thing is that whatever I put in both `extensions.zotero.translators.better-bibtex.citekeyFormat` and `extensions.zotero.translators.better-bibtex.citekeyFormatEdit`, it just resets to `auth.lower + shorttitle(3,3) + year` when I run Zotero. Unless I manually update the field via GUI, the change doesn't persist. This happens when I update the fields either or both `prefs.js` or `user.js`.

**Reply by retorquere** [reply]
And this is on the newest version?

**Reply by okomestudio** [reply]
Yes. Version 6.7.240. Nothing has changed, as far as I can see on the issue.

**Reply by retorquere** [reply]
Please open an issue so we may diagnose what's happening.

**Reply by okomestudio** [reply]
https://github.com/retorquere/zotero-better-bibtex/issues/3013
